core/domain/state_domain.py

```diff
@@ -1428,18 +1429,28 @@ class WrittenTranslation(python_utils.OBJECT):
     """Value object representing a written translation for a content."""
 
     DATA_FORMAT_HTML = 'html'
-    DATA_FORMAT_UNICODE = 'unicode'
+    DATA_FORMAT_UNICODE_STRING = 'unicode'
+    DATA_FORMAT_SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    DATA_FORMAT_SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+    DATA_FORMAT_TO_TRANSLATABLE_OBJ_CLASS_NAME = {
```

Comment:
[Optional]
```suggestion
    DATA_FORMAT_TO_TRANSLATABLE_OBJ_TYPE = {
```

Just to have one terminology for one thing in the codebase.

Reply:
Ah, there's a specific reason I used "class name" here. In this function, I'm not using the translatable object as a conceptual object (which has a type), I'm just trying to retrieve the function within the corresponding class that validates the "inner value". So I think it's more accurate in the context of this usage to be talking about the object class because I want a function that's part of that class and I don't actually care about the object type as a whole.

Reply:
You mean in this file we are just using the classmethod of the class to normalize the value without creating an "object", right? (Sorry if I misunderstood your comment)

I'm assuming OBJECT_TYPE to say `type(class) i.e, the string` here and it will match with the param name we pass in the calling function i.e,  `get_object_class(cls, obj_type)`

Reply:
Correct, that's what I meant by what I said.

Eh, I don't really mind changing it if you feel strongly. But can you please give me a full list of changes before I do that, so that the CI checks only need to run one more time?

Reply:
[Optional] I think it would be nice to build this term here which can be followed and understood in the translation_obj_registry.py file.

Reply:
Done!